contracts/MetaFactory.sol

```diff
@@ -9,315 +9,82 @@ import "@openzeppelin/contracts/utils/Address.sol";
-    /// @param roleModuleMembers Indexes of the modules to be granted each role
-    /// @param newContractAddresses Array of addresses of the newly created contracts
-    function addModuleRoles(
-        string[] memory roles,
-        uint256[][] memory roleModuleMembers,
-        address[] memory newContractAddresses
-    ) private {     
-        uint256 newMembersLength = roleModuleMembers.length;
-        address[][] memory newMembers = new address[][](newMembersLength);
-        for (uint256 i; i < newMembersLength; ) {
-            uint256 newMembersInnerLength = roleModuleMembers[i].length;
-            address[] memory newMembersInner = new address[](newMembersInnerLength);
-            for (uint256 j; j < newMembersInnerLength; ) {
-                newMembersInner[j] = newContractAddresses[
-                    roleModuleMembers[i][j]
-                ];
-                unchecked {
-                    j++;
-                }
-            }
-            newMembers[i] = newMembersInner;
-            unchecked {
-                i++;
-            }
-        }
-
-        bytes memory data = abi.encodeWithSignature(
-            "grantRoles(string[],address[][])",
-            roles,
-            newMembers
-        );
-
-        address[] memory targetArray = new address[](1);
-        uint256[] memory valuesArray = new uint256[](1);
-        bytes[] memory dataArray = new bytes[](1);
-
-        targetArray[0] = newContractAddresses[1];
-        valuesArray[0] = 0;
-        dataArray[0] = data;
-
-        IDAO(newContractAddresses[0]).execute(
-            targetArray,
-            valuesArray,
-            dataArray
-        );
-    }
-
-    /// @notice Returns whether a given interface ID is supported
-    /// @param interfaceId An interface ID bytes4 as defined by ERC-165
-    /// @return bool Indicates whether the interface is supported
-    function supportsInterface(bytes4 interfaceId)
-        public
-        view
-        virtual
-        override
-        returns (bool)
-    {
-        return
-            interfaceId == type(IMetaFactory).interfaceId ||
-            super.supportsInterface(interfaceId);
+  /// @notice Creates a DAO, Access Control, and any modules specified
+  /// @param daoFactory The address of the DAO factory
+  /// @param createDAOParams The struct of parameters used for creating the DAO and Access Control contracts
+  /// @param targets An array of addresses to target for the function calls
+  /// @param values An array of ether values to send with the function calls
+  /// @param calldatas An array of bytes defining the function calls
+  function createDAOAndExecute(
+    address daoFactory,
+    IDAOFactory.CreateDAOParams memory createDAOParams,
+    address[] calldata targets,
+    uint256[] calldata values,
+    bytes[] calldata calldatas
+  ) external {
+    createDAO(daoFactory, createDAOParams);
+    execute(targets, values, calldatas);
+  }
+
+  function createDAO(
```

Comment:
Please add some NatSpec comments to this function (since there are comments on the other `internal` function...)